Entitas/Entitas/CodeGenerator/Generators/ComponentExtensionsGenerator.cs

```diff
@@ -135,7 +195,7 @@ public class ComponentExtensionsGenerator : IComponentCodeGenerator {
         public Entity Add$Name($typedArgs) {
             var component = CreateComponent<$Type>($Ids.$Name);
 $assign
-            return AddComponent($Ids.$Name, component);
+            return (Entity)AddComponent($Ids.$Name, component);
```

Comment:
Because the `IEntity` interface declares `IEntity AddComponent(int index, IComponent component);` this needs to be casted. It will always succeed.